virtualfish/virtual.fish

```diff
@@ -47,10 +47,27 @@ function __vf_activate --description "Activate a virtualenv"
         echo "You need to specify a virtualenv."
         return 1
     end
-    if not [ -d $VIRTUALFISH_HOME/$argv[1] ]
-        echo "The virtualenv $argv[1] does not exist."
-        echo "You can create it with `vf new $argv[1]`."
-        return 2
+
+    if echo $argv[1] | grep -q /
+        # The argument contains slashes: user requested to activate
+        # virtualenv outside VIRTUALFISH_HOME.
+        set -f venv_path $PWD/$argv[1]
+
+        if not [ -d $venv_path ]
+            echo "Relative path $argv[1] does not exist."
+            return 2
+        else if not [ -e $venv_path/bin/activate ]
+            echo "Relative path $argv[1] does not appear to be a virtualenv."
+            return 2
+        end
+    else
+        set -f venv_path $VIRTUALFISH_HOME/$argv[1]
+
+        if not [ -d $venv_path ]
+            echo "The virtualenv $argv[1] does not exist."
+            echo "You can create it with `vf new $argv[1]`."
+            return 2
+        end
     end
```

Comment:
Because it doesn't make much sense to define the `venv_path` variable if the path doesn't exist — and thus we are about to exit with an error — I think I would invert order of these things, perhaps like this:

```fish
        if not [ -d $VIRTUALFISH_HOME/$argv[1] ]
            echo "The virtualenv $argv[1] does not exist."
            echo "You can create it with `vf new $argv[1]`."
            return 2
        else
            set -f venv_path $VIRTUALFISH_HOME/$argv[1]
        end
```